src/Microsoft.Identity.Web/TokenCacheProviders/MsalAbstractTokenCacheProvider.cs

```diff
@@ -13,6 +20,28 @@ namespace Microsoft.Identity.Web.TokenCacheProviders
     /// <seealso cref="Microsoft.Identity.Web.TokenCacheProviders.IMsalTokenCacheProvider" />
     public abstract class MsalAbstractTokenCacheProvider : IMsalTokenCacheProvider
     {
+        private readonly IServiceProvider? _serviceProvider;
+        private readonly IDataProtector? _protector;
+        private readonly string _defaultPurpose = "msal_cache";
+
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="serviceProvider">Service provider.</param>
+        protected MsalAbstractTokenCacheProvider(IServiceProvider? serviceProvider = null)
+        {
+            _serviceProvider = serviceProvider;
+            if (_serviceProvider != null)
+            {
+                IDataProtectionProvider? dataProtectionProvider = _serviceProvider.GetService(typeof(IDataProtectionProvider)) as IDataProtectionProvider;
+#if DOTNET_472 || DOTNET_462
+                _protector = dataProtectionProvider?.Create(_defaultPurpose);
```

Comment:
Curious: why not require the IDataProtector directly instead of the IServiceProvider? 

Context:
deciding whether to use data protection based on nullity of ServiceProvider does not IMHO show clear intent. (also not documented on the xmldoc)

Reply:
yes, have updated the xml comment.